policy/compressor_test.go

```diff
@@ -81,16 +91,51 @@ func TestDynamicCompressor(t *testing.T) {
 		resolution: Resolution{Window: 10 * time.Minute, Precision: xtime.Minute},
 		retention:  Retention(48 * time.Hour),
 	}
-	expectedID := int64(len(compressorPolicies) + 1)
-	mock.insert(newPolicy, expectedID)
-	mock.flush()
-	id, ok := c.Get(newPolicy)
+	id := int64(len(compressorPolicies) + 1)
+	policies[newPolicy] = id
+	proto = compressorProtoFromPolicies(policies)
+	vw.Update(mem.NewValue(2, proto))
+
+	for i := 0; i < 100; i++ {
+		if _, ok := c.Get(newPolicy); ok {
+			break
+		}
+		time.Sleep(10 * time.Millisecond)
+	}
+
+	actualID, ok := c.Get(newPolicy)
 	assert.True(t, ok)
-	assert.Equal(t, expectedID, id)
+	assert.Equal(t, id, actualID)
 
 	// Remove the new policy
-	mock.remove(newPolicy)
-	mock.flush()
+	delete(policies, newPolicy)
+	proto = compressorProtoFromPolicies(policies)
+	vw.Update(mem.NewValue(3, proto))
+
+	for i := 0; i < 100; i++ {
+		if _, ok := c.Get(newPolicy); !ok {
+			break
+		}
+		time.Sleep(10 * time.Millisecond)
+	}
 	_, ok = c.Get(newPolicy)
 	assert.False(t, ok)
 }
+
+func compressorProtoFromPolicies(policies map[Policy]int64) proto.Message {
+	activePolicies := &schema.ActivePolicies{}
```

Comment:
this function seems to be useful outside of test as well, e.g can be used in the cmd tool to save ActivePolicies, make it a util function?